v2/pkg/engine/plan/abstract_selection_rewriter.go

```diff
@@ -64,7 +64,8 @@ type fieldSelectionRewriter struct {
 	upstreamDefinition *ast.Document
 	dsConfiguration    DataSource
 
-	skipFieldRefs []int
+	skipFieldRefs    []int
+	enforceRewriting bool
```

Comment:
What do you think about naming it like `alwaysFlattenFragments` or just `alwaysRewrite`